packages/stark-ui/src/modules/pretty-print/components/pretty-print.component.spec.ts

```diff
@@ -342,100 +374,124 @@ describe("PrettyPrintComponent", () => {
+		describe("Typescript", () => {
 			beforeEach(() => {
-				hostComponent.data = rawTypeScriptData;
+				hostComponent.data = rawTypescriptData;
 				hostComponent.format = "typescript";
 				hostFixture.detectChanges();
 			});
 
 			it(shouldHaveInputs, () => {
-				expect(component.data).toBe(rawTypeScriptData);
+				expect(component.data).toBe(rawTypescriptData);
 				expect(component.format).toBe("typescript");
 				expect(component.enableHighlighting).toBeUndefined();
 			});
 
-			it("should leave the raw TypeScript data unformatted (TS formatting not yet supported)", () => {
-				expect(component.prettyString).toBe(rawTypeScriptData);
+			it("should nicely format raw typescript data", () => {
+				expect(component.prettyString).toBe(formattedTypescriptData);
 
 				const preElement: HTMLPreElement | null = <HTMLPreElement>hostFixture.nativeElement.querySelector("pre");
 				expect(preElement).not.toBeNull();
-				/* tslint:disable: no-duplicate-string */
-				expect(preElement.innerHTML).toContain("function calculateData(seed:any, operationFn:Function):any");
-				/* tslint:enable: no-duplicate-string */
+				expect(preElement.innerHTML).toContain("function calculateData(seed: any, operationFn: Function): any {");
+				expect(preElement.innerHTML).toContain("var data: any = operationFn(seed);");
+				expect(preElement.innerHTML).toContain("if (!data) {");
+			});
+
+			it("should simply display the unformatted raw typescript data in case it is not valid typescript", () => {
+				const invalidRawTypescriptData: string = rawTypescriptData + "}";
+
+				hostComponent.data = invalidRawTypescriptData;
+				hostComponent.format = "typescript";
+				hostFixture.detectChanges();
+
+				expect(component.prettyString).toBe(invalidRawTypescriptData);
+
+				const preElement: HTMLPreElement | null = <HTMLPreElement>hostFixture.nativeElement.querySelector("pre");
+				expect(preElement).not.toBeNull();
+				expect(preElement.innerHTML).toContain("if (!data){data = 'could not calculate data';}return data;");
 			});
 		});
 
 		describe("Undefined format", () => {
 			beforeEach(() => {
-				hostComponent.data = rawTypeScriptData;
+				hostComponent.data = rawTypescriptData;
 				hostComponent.format = <any>undefined;
 				hostFixture.detectChanges();
 			});
 
 			it(shouldHaveInputs, () => {
-				expect(component.data).toBe(rawTypeScriptData);
+				expect(component.data).toBe(rawTypescriptData);
 				expect(component.format).toBeUndefined();
 				expect(component.enableHighlighting).toBeUndefined();
-			});
+			}); // the same test is performed with an unknown format
 
-			/* tslint:disable: no-identical-functions */ // the same test is performed an unkowm format
-			it("should leave the raw data unformatted when the format is not defined", () => {
-				expect(component.prettyString).toBe(rawTypeScriptData);
+			/* tslint:disable: no-identical-functions */ it("should leave the raw data unformatted when the format is not defined", () => {
```

Comment:
The format of these 2 comment lines is weird and of course is due to Prettier. I suggest to change them like this:

```
// the same test is performed with an unknown format
// tslint:disable-next-line: no-identical-functions
it (......)
```